app/dose-event-fab/dose-event-fab.component.ts

```diff
@@ -0,0 +1,27 @@
+import {Component, Input, Type} from "@angular/core";
+import {DoseEvent} from "../dose-event/dose-event";
+import {DOSE_EVENTS} from "../dose-event-service/dose-event-mocks";
+import {DoseEventSliderComponent} from "../dose-event-slider/dose-event-slider.component";
+
+/**
+ * DoseEventFab for rendering a dose-events.
+ */
+@Component(
+    {
+        selector: "dose-event-fab",
+        templateUrl: "build/dose-event-fab/dose-event-fab.component.html"
+    }
+)
+export class DoseEventFabComponent {
+
+    @Input()
+    public doseEvents: Array<DoseEvent>;
+
+    takeAll() {
+        for (let dose of this.doseEvents) {
+            dose.action = 'TAKEN';
+        }
+
+    }
+
+}
```

Comment:
This DoseEventFabComponent  class looks correct now, (except you used single quotes so lint will fail).